lib/usd/translators/mayaReferenceUpdater.cpp

```diff
@@ -205,30 +205,47 @@ UsdMayaPrimUpdater::PushCopySpecs PxrUsdTranslators_MayaReferenceUpdater::pushCo
     (*dstEditRouter)(routerContext, routingData);
 
     // Retrieve the destination layer and prim path from the routing data.
-    auto dstLayerStr = findValue(routingData, TfToken("layer"));
-    if (!TF_VERIFY(!dstLayerStr.empty())) {
+    auto cacheDstLayerStr = findValue(routingData, TfToken("layer"));
+    if (!TF_VERIFY(!cacheDstLayerStr.empty())) {
         return PushCopySpecs::Failed;
     }
-    auto dstPathStr = findValue(routingData, TfToken("path"));
-    if (!TF_VERIFY(!dstPathStr.empty())) {
+    auto cacheDstPathStr = findValue(routingData, TfToken("path"));
+    if (!TF_VERIFY(!cacheDstPathStr.empty())) {
         return PushCopySpecs::Failed;
     }
 
-    auto dstLayer = SdfLayer::FindOrOpen(dstLayerStr);
-    if (!TF_VERIFY(dstLayer)) {
+    // Copy transform changes that came from the Maya transform node into the
+    // Maya reference prim.  The Maya transform node changes have already been
+    // exported into the temporary layer as a transform prim, which is our
+    // source.  The destination prim in the stage is the Maya reference prim.
+    auto srcPrim = srcStage->GetPrimAtPath(srcSdfPath);
+    if (TF_VERIFY(UsdGeomXformable(srcPrim))) {
+        auto dstPrim = dstStage->GetPrimAtPath(dstSdfPath);
+        if (TF_VERIFY(UsdGeomXformable(dstPrim))) {
+            // The Maya transform that corresponds to the Maya reference prim
+            // only has its transform attributes unlocked.  Bring any transform
+            // attribute edits over to the Maya reference prim.
+            MayaUsdUtils::MergePrimsOptions options;
+            options.ignoreUpperLayerOpinions = true;
+            MayaUsdUtils::mergePrims(
```

Comment:
mergePrims returns a boolean for success. We might at least want to warn on failure. (IDK if the failure cases already log something or not, it might depend on the cause of failure.)